src/fluent.rs

```diff
@@ -160,6 +196,36 @@ impl Localizer {
     }
 }
 
+pub trait MessageKey {
+    fn key(&self) -> &str;
+
+    fn attribute(&self) -> Option<&str> {
+        None
+    }
+}
+
+impl MessageKey for str {
+    fn key(&self) -> &str {
+        self
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct MessageAttribute<'key, 'attribute> {
+    pub key: &'key str,
+    pub attribute: &'attribute str,
+}
+
+impl<'key, 'attribute> MessageKey for MessageAttribute<'key, 'attribute> {
+    fn key(&self) -> &str {
+        self.key
+    }
+
+    fn attribute(&self) -> Option<&str> {
+        Some(self.attribute)
+    }
+}
```

Comment:
Maybe this ought to be over with the tera stuff? Is this intended for tera?

If this is for Tera, won't the lifetimes be the same for both the attribute and the key?

Reply:
You could probably get away with using one lifetime in 99% of cases, but it's gonna be annoying for the remaining 1%. Personally, since the user will not have to specify these manually ever, I thought I was fine to go with two different lifetimes, since it does not really raise usage complexity.

Reply:
Also, no, other users of this crate (not using Tera) could also use this as well to use message attributes instead of just plain messages.